htdocs/config.php

```diff
@@ -5,14 +5,14 @@
 define('TIME', (int)microtimeSec(MICRO_TIME));
 
 // Repository paths
-define('ROOT', __DIR__ . '/../');
-define('LIB', ROOT.'lib/');
-define('ENGINE', ROOT.'engine/');
-define('WWW', ROOT.'htdocs/');
-define('UPLOAD', WWW.'upload/');
-define('ADMIN', ROOT.'admin/');
-define('TOOLS', ROOT.'tools/');
-define('CONFIG', ROOT.'config/');
+const ROOT = '/smr/';
```

Comment:
Hard-coding the `ROOT` directory will break the non-docker instance that is running on port 444 of the live server. I was mainly keeping this up to test the performance degradation in the docker instance, but now that it is resolved (see #297), perhaps there is no longer any reason to keep that running.

Is the reason for the change that `__DIR__` isn't known at compile time? If so, I'd be willing to endure the additional cost of a few `define`'s if it means we get to keep the flexibility of `__DIR__`. What do you think?

Reply:
I actually thought this was in a `config.specific.sample.php` still (it popped up in a bunch of specific files I had locally, and I didn't spot this wasn't one) and therefore that having the default be `/smr/` made the most sense as it allowed `const` and also to use the defaults as-is when running via docker.  As for `__DIR__` in a const expr I don't know as I didn't check because of those reasons

Reply:
Do you want me to give it a try and push if `__DIR__` works?

Reply:
Yeah that'd be great, otherwise it's probably gonna be a couple days until I can check it out